src/filters.rs

```diff
@@ -0,0 +1,1115 @@
+use std::collections::{HashMap, HashSet};
+
+use zksync_basic_types::{H160, H256, U256, U64};
+use zksync_types::api::{BlockNumber, Log};
+use zksync_web3_decl::types::FilterChanges;
+
+/// Specifies a filter type
+#[derive(Debug, Clone, PartialEq)]
+pub enum FilterType {
+    /// A filter for block information
+    Block(BlockFilter),
+    /// A filter for log information.
+    /// This is [Box] to ensure the enum invariants are similar size.
+    Log(Box<LogFilter>),
```

Comment:
I'm a bit on the fence with this warning. I get the utility of it and in general agree to it, but this doesn't seem to be that future safe if for some reason _all_ invariants become huge. Admittedly in our case we know this won't happen, but I just found it a bit weird :D 

![image](https://github.com/matter-labs/era-test-node/assets/1564843/a3ecf6a5-98f7-4cb6-ac63-ea910c0b0e46)

Reply:
I'm still a newbie with Rust, so I just went down a rabbit hole of variant sizes and `Box<>`... at this point I'm deferring the almighty clippy, but agreed that this is weird

Reply:
It makes total sense what clippy suggests, since the size of an enum/union is as big as the biggest invariant. In our case each `Log` object would be `272 bytes` even if we store 100x `Block` invariant which is a mere `24 bytes`. This can cause a very high memory usage for no good reason when storing a significant number of objects. Therefore, clippy suggests us to allocate `LogFilter` on the heap where it will only occupy `usize` worth of size for the pointer - mitigating the issue.

But..... I know for a fact this will never be significant for us, and is almost certainly a premature optimization at best and a code smell at worst (since this isn't intuitive). Hence, my apprehension, but then again I'm also not a huge fan of adding ignores to linter thus I went with the comment explanation.